ycmd/tests/get_completions_test.py

```diff
@@ -330,9 +1129,17 @@ def GetCompletions_CsCompleter_NonForcedReturnsNoResults_test():
                                   column_num = 21,
                                   force_semantic = False,
                                   query = 'Date' )
-  results = app.post_json( '/completions', completion_data ).json[ 'completions' ]
-
-  assert_that( results, empty() )
+  results = app.post_json( '/completions', completion_data ).json
+
+  # there are no semantic completions. However, we fall back to identifier
+  # completer in this case.
```

Comment:
Hmmm. The following tests seem to be explicitly testing "non-forced" semantic completions (or the interaction of forced and non-forced in the latter case). So i'm tempted to leave them as-is, so as not to interfere with the idea behind the original test.

```python
def GetCompletions_CsCompleter_NonForcedReturnsNoResults_test():
def GetCompletions_CsCompleter_ForcedDividesCache_test():
```

These ones test exceptions from the completer, and in particular the common-ish case where there is no .ycm_extra_conf (and therefore not compatible with my generic `GetCompletions_RunTest` function), but i think similar tests do exist, so i've reverted the changes:

```python
GetCompletions_ClangCompleter_ExceptionWhenNoFlagsFromExtraConf_test
GetCompletions_ClangCompleter_ExceptionWhenNoFlagsFromExtraConf_ID_test
```

What do you think?